test/e2e/parallel/space_cleanup_test.go

```diff
@@ -104,7 +104,8 @@ func TestSpaceAndSpaceBindingCleanup(t *testing.T) {
 }
 
 func setupForSpaceBindingCleanupTest(t *testing.T, awaitilities wait.Awaitilities, targetMember *wait.MemberAwaitility, murName, spaceName string) (*toolchainv1alpha1.Space, *toolchainv1alpha1.UserSignup, *toolchainv1alpha1.SpaceBinding) {
-	space := CreateAndVerifySpace(t, awaitilities, WithTierName("appstudio"), WithTargetCluster(targetMember), WithName(spaceName))
+	space, _, _ := CreateSpace(t, awaitilities, WithTierName("appstudio"), WithTargetCluster(targetMember), WithName(spaceName), WithCreatorName(murName))
+	VerifyResourcesProvisionedForSpace(t, awaitilities, space.Name, wait.UntilSpaceHasCreatorLabel(murName), wait.UntilSpaceHasStatusTargetCluster(targetMember.ClusterName))
```

Comment:
removed the explicit setting of `CreatorName` label in 2b2f705
 
> in addition - why did you replace the CreateAndVerifySpace with two calls?

that's related to the use of `VerifyResourcesProvisionedForSpace` vs `VerifySpaceRelatedResources`, but that change will be completed in a subsequent PR. If it's not critical, I suggest we keep this change for now ;)

Reply:
could you please share the reasoning/explanation for it, because I'm really confused now.

Reply:
My confusion is mainly because this introduces some changes in the code that doesn't make sense in the context of this PR, but now you are saying that it's only the first half of the changes that you need to make and that there will be another set of the changes in subsequent PR, which wasn't mentioned anywhere. 

I'm curious what is the change that it cannot be done at once because now I need to keep in mind what was changed here to and should be addressed in the following PR. 
Is it the size of the changes? if so then what is the main difference compared to the current code and the way we verify the resources provisioned for Space. Because I don't see any big difference (but it's maybe because it's Monday)

Please, don't take my comments wrong - I just got lost, so I'm asking for more hints...